src/gui/modals/FileDialog.cpp

```diff
@@ -66,6 +89,25 @@ FileDialog::FileDialog( QWidget *parent, const QString &caption,
 		urls << QUrl::fromLocalFile( volumesDir.absolutePath() );
 #endif
 
+#ifdef LMMS_BUILD_LINUX
+
+	// FileSystem types : https://www.javatpoint.com/linux-file-system
+	QStringList usableFileSystems = {"ext", "ext2", "ext3", "ext4", "jfs", "reiserfs"};
+
+	foreach (QStorageInfo storage, QStorageInfo::mountedVolumes())
+	{
+		storage.refresh();
+
+		if (usableFileSystems.contains(QString(storage.fileSystemType()), Qt::CaseInsensitive))
+		{
+			if (storage.isValid() && storage.isReady())
+			{
+				urls << QUrl::fromLocalFile(storage.displayName());	
```

Comment:
This line should be changed to:
```
urls << QUrl::fromLocalFile(storage.rootPath());
```
Otherwise it won't work with partitions that have names/labels. Here's a snippet of some debug output for my filesystem:
```
Display name:  "/data/linux" , Name:  "" , Root Path:  "/data/linux"
Display name:  "Linux-Homes" , Name:  "Linux-Homes" , Root Path:  "/home"
Display name:  "usb-2.0-data" , Name:  "usb-2.0-data" , Root Path:  "/mnt"
```
My `home` folder is mounted on a separate `ext4` partition that is named "Linux-Homes". If `displayName` is added to the lists of URLs then "Linux-Homes" will be shown in the sidebar. This looks nice but unfortunately it does not work when I click on it. Changing the line as described above leads to `/home`  being displayed in the sidebar. This does not look as nice but it's working.

The partition `/data/linux` does not have a name. Hence the display name and the technical root path are the same and it also works with the display name in the URL list.

Unfortunately it does not seem to be possible to differentiate in the sidebar between the name that should be displayed and the technical path where the data resides.